lib/pundit.rb

```diff
@@ -218,11 +218,12 @@ def authorize(record, query = nil, policy_class: nil)
 
     @_pundit_policy_authorized = true
 
-    policy = policy_class ? policy_class.new(pundit_user, record) : policy(record)
+    actual_record = record.is_a?(Array) ? record.last : record
+    policy = policy_class ? policy_class.new(pundit_user, actual_record) : policy(record)
 
     raise NotAuthorizedError, query: query, record: record, policy: policy unless policy.public_send(query)
```

Comment:
This needs to have the same change, otherwise the error message reads "not allowed to update this Array"

```suggestion
    raise NotAuthorizedError, query: query, record: actual_record, policy: policy unless policy.public_send(query)
```